7_drivers_munge/src/GCM_driver_utils.R

```diff
@@ -154,36 +155,73 @@ get_lake_cell_tile_xwalk <- function(lake_centroids, grid_cells, cell_tile_xwalk
   return(lake_cells_tiles_join)
 }
 
-#' @title Map the query tiles and cells
+#' @title Match lakes to the queried cells that returned data.
+#' @description Using lake centroids, this spatially matches lakes to query
+#' cells that returned data, based on which cell centroid is closest to each
+#' lake centroid. The output of this function will only differ from the output
+#' of `get_lake_cell_tile_spatial_xwalk()` for those lakes that fell within
+#' cells that did not return data
+#' @param lake_centroids an `sf` object of points representing the lake
+#' centroids.
+#' @param query_cell_centroids  an `sf` object of points representing the
+#' grid cell centroids for queried cells. Must contain a `cell_no` column
+#' which has the id of each of the cell centroids
+#' @param cell_status a table with one row per query cell, with columns for
+#' `tile_no` and each gcm (`{gcm}_missing_data`), indicating whether or not the cell
+#' is missing data for any variable for that gcm. The table also includes a
+#' final column `n_gcm_missing_data` with the total count of gcms that are missing
+#' data for that cell. This table is used here to filter the grid cell
+#' centroids to only those that returned data
+#' @return An output table with the fields `site_id`, `state`, `cell_no` and
+#' `tile_no`
+get_lake_cell_tile_data_xwalk <- function(lake_centroids, query_cell_centroids, cell_status) {
+  # determine which cells returned data for all 6 GCMs
+  cells_with_data <- cell_status %>%
+    filter(n_gcm_missing_data == 0) %>%
+    pull(cell_no)
+
+  # filter the cell centroids to only those cells with data
+  cell_centroids_with_data <- query_cell_centroids %>%
+    filter(cell_no %in% cells_with_data)
+
+  # match each lake to a cell that returned data based on
+  # which cell centroid is closest to each lake centroid
+  lake_cells_with_data_tiles_join <- lake_centroids %>%
+    st_join(cell_centroids_with_data, join=st_nearest_feature, left=TRUE) %>%
```

Comment:
Okay I dug into this a bit yesterday afternoon, and I figured out a way to only match lakes that fall in NA cells to non-NA cells within the same row. In this figure, the lime green dots = the original matching based on `st_nearest_feature` and the white dots with black borders = the revised matching if we restrict it to non-NA cells within the same row:
![image](https://user-images.githubusercontent.com/54007288/155752920-ee831e8a-b727-4d06-ad13-3e7ae58e3af3.png)
I think there are 2 things we need to consider:
1) what if there are no non-NA cells in the same grid row as lakes that fall into NA cells? That's not the case here, but could be when we expand the footprint. I think we could add in an if statement that reverts back to the use of `st_nearest_feature` to match cells across rows if that is the case.
2) how far are we willing to go in the x-direction? For lakes in the northernmost 2 missing cells in tile 78, the closest non-NA cell within the same row is 3-4 cells to the left. Are we okay with the matched cell being far away, or do we want to set some threshold beyond which we'd rather allow the match to be across rows?

Reply:
Great points to consider! How far away east/west would you need to be before an adjacent north/south cell would be a better estimate? 2 cells? 5? 10? Thinking through it in an area I know: Richland Center is ~1 deg west of Madison and Delafield is ~1 deg east of Madison. ~1 deg in longitude looks like 3 cells or so. I would say those locations experience similar-ish climates to Madison but I definitely wouldn't think we could go any further out than that.

Either way we would A) probably a good idea to make this an easily configurable feature so we can change in the future if we need to, and B) solve both no-NA cells in the same grid row with the same solution bc the nearest x-direction cell (none) would be greater than whatever we set as our max.

Reply:
okay, cool. I'll work on a solution that is configurable, then. Thanks for the input!

Reply:
Ok I think I have this working, with an `x_buffer` value of 3 cells. Here are the current non-nan cells that match to lakes within the cells that are missing data, colored according to whether or not the matched cell has the same y-value as the cell that contains the lake.
![image](https://user-images.githubusercontent.com/54007288/156009629-56ec2731-1b3e-4a48-92bc-e2ad43b0de1f.png)

You can see that only one matched cell has a y-value that is different, and that's for the right-most NaN cell in the top row of tile 78. Here in white is the cell it would match if there were no `x_buffer`, and in purple the cell it matches with the `x_buffer` specified as 3:
![image](https://user-images.githubusercontent.com/54007288/156009565-85c2a47e-3381-4499-b2b7-b1aa01b0b66f.png)

Reply:
I'll look at the code now, but the result from adding these criteria seem appropriate. I think we should make sure to log this change in our metadata for the data release. So, maybe we need to create an issue over in https://github.com/USGS-R/laketemp-projections-data-release with the description of how NA cells are matched.

Reply:
Done - see [this issue](https://github.com/USGS-R/laketemp-projections-data-release/issues/3)